File: apps/web/lib/api/ai.ts
import { apiFetch } from "./client";
import { getToken } from "@/lib/auth/session";

function authHeaders(): HeadersInit {
  const token = getToken();
  return token ? { Authorization: `Bearer ${token}` } : {};
}

export interface AIStatus {
  configured: boolean;
  provider: string;
}

export function getAIStatus(): Promise<AIStatus> {
  return apiFetch<AIStatus>("/ai/status");
}

export interface ChatMessageIn {
  role: "user" | "assistant";
  content: string;
}

export interface ToolCallSummary {
  name: string;
  arguments: Record<string, unknown>;
}

export interface ChatResponse {
  reply: string;
  tool_calls: ToolCallSummary[];
}

export function sendChatMessage(
  message: string,
  history: ChatMessageIn[],
  businessId?: number
): Promise<ChatResponse> {
  return apiFetch<ChatResponse>("/ai/chat", {
    method: "POST",
    headers: authHeaders(),
    body: JSON.stringify({ message, history, business_id: businessId ?? null }),
  });
}

File: apps/api/app/schemas/ai.py
from typing import Any, Literal

from pydantic import BaseModel, Field


class AIStatus(BaseModel):
    configured: bool
    provider: str


class ChatMessageIn(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(min_length=1, max_length=4000)


class ChatRequest(BaseModel):
    message: str = Field(min_length=1, max_length=2000)
    history: list[ChatMessageIn] = Field(default_factory=list, max_length=20)
    business_id: int | None = None


class ToolCallSummary(BaseModel):
    name: str
    arguments: dict[str, Any]


class ChatResponse(BaseModel):
    reply: str
    tool_calls: list[ToolCallSummary]
